src/accountsdb/cache.zig

```diff
@@ -0,0 +1,365 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+const builtin = @import("builtin");
+const zstd = @import("zstd");
+
+const Account = sig.core.Account;
+const Slot = sig.core.time.Slot;
+const Pubkey = sig.core.pubkey.Pubkey;
+
+const LruCacheCustom = sig.common.lru.LruCacheCustom;
+
+/// Stores read-only in-memory copies of commonly used accounts
+pub const AccountsCache = struct {
+    /// Atomically refcounted account
+    pub const CachedAccount = struct {
+        account: Account,
+        /// represents whether this account has been mutated, and shouldn't progress to newer slots
+        // TODO: when we start mutating accounts, make sure to set this field
+        is_dirty: std.atomic.Value(bool) = .{ .raw = false },
```

Comment:
Every time you want to mark an account dirty, you need to get it from the Lru. This simultaneously marks the account as the most recently used item, and also as a stale item at the same time. Something feels off about that.